### PR TITLE
Initial documentation

### LINKED ISSUE #11: Initial documentation
We need to release a proper documentation for 1.0

- [x] Installation
- [x] Usage
- [x] Image descriptor format
- [x] Conventions
- [ ] Configuration file

### PR DESCRIPTION
**Pull request by goldmann**
Added information about the descriptor format

Fixes #11

### CODE REVIEW

File: docs/descriptor.rst [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+---------------------
+
+Similar to labels -- we can specify environment variables that should be
+present in the container after running the image. We provide ``envs``
+section for this.
+
+Environment variables can be divided into two types:
+
+1. **Information environment variables** -- these are set and available in
+   the image. This type of envirpnment variables provide information to
```
Comment (dbecvarik) [inline docs/descriptor.rst]: typo envirpnment ->  environment

File: docs/descriptor.rst [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Environment variables
+---------------------
+
+Similar to labels -- we can specify environment variables that should be
+present in the container after running the image. We provide ``envs``
+section for this.
+
+Environment variables can be divided into two types:
+
+1. **Information environment variables** -- these are set and available in
```
Comment (dbecvarik) [inline docs/descriptor.rst]: I would just skip the first sentence here and start with. These env... variables provide information... - maybe mention docker inspect to see them somewhere?

File: docs/descriptor.rst [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+   the image. This type of envirpnment variables provide information to
+   the image consumer. In most cases such environment variables *should not*
+   be modified.
+2. **Configuration environment variables** -- this type of variables are
```
Comment (dbecvarik) [inline docs/descriptor.rst]: I think, that we miss there most important thing - that theese variable are used to configure features of container during runtime.

File: docs/descriptor.rst [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+.. code:: yaml
+
+    name: "jboss-eap-7/eap70-openshift"
+    version: "1.4"
+    from: "jboss-eap-7-tech-preview/eap70:1.2"
+    description: "Red Hat JBoss Enterprise Application 7.0 - An application platform for hosting your apps that provides an innovative modular, cloud-ready architecture, powerful management and automation, and world class developer productivity."
+    user: "alice"
+
+Labels
```
Comment (dbecvarik) [inline docs/descriptor.rst]: its weird going from glossary terms to chapters. It feels inconsistent.

File: docs/descriptor.rst [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        - mongodb24-mongo-java-driver
+        - postgresql-jdbc
+        - mysql-connector-java
+        - maven
+        - hostname
+
+Artifacts
+---------
+
+It's not a common case where your image won't require external artifacts.
```
Comment (dbecvarik) [inline docs/descriptor.rst]: Its common for images to require external artifacts - sounds clearer than double negation for me.

File: docs/descriptor.rst [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+In most cases you will want to add files into the image and use them in
+the image building process.
+
+Artifacts section is meant exactly for this. *Concreate will automatically
+fetch any artifacts* specified in this section
```
Comment (dbecvarik) [inline docs/descriptor.rst]: fetch all artifacts?

File: docs/descriptor.rst [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    user: "alice"
+
+Labels
+------
+
+Every image can include labels. Concreate makes it easy to do so with the ``labels`` section.
+
+.. code:: yaml
+
+    labels:
```
Comment (dbecvarik) [inline docs/descriptor.rst]: I would mention and link https://github.com/projectatomic/ContainerApplicationGenericLabels as a resource to study/see.

File: docs/descriptor.rst (hunk 2 above)
Comment (goldmann) [reply docs/descriptor.rst]: Fro now on we will make the tool not tied to any implementation, including Docker.

File: docs/descriptor.rst (hunk 4 above)
Comment (goldmann) [reply docs/descriptor.rst]: The difference is that the "General information" section contains multiple keys, whereas chapters/section talk about a single one.

File: docs/descriptor.rst (hunk 7 above)
Comment (goldmann) [reply docs/descriptor.rst]: Good idea.

File: docs/descriptor.rst (hunk 1 above)
Comment (goldmann) [reply docs/descriptor.rst]: Good catch.

File: docs/descriptor.rst (hunk 3 above)
Comment (goldmann) [reply docs/descriptor.rst]: I'll clarify it.

File: docs/descriptor.rst (hunk 5 above)
Comment (goldmann) [reply docs/descriptor.rst]: I'll change it.

File: docs/descriptor.rst (hunk 6 above)
Comment (goldmann) [reply docs/descriptor.rst]: Yep.

**Comment by goldmann** [comment]
There is one thing that I haven't mentioned yet - I'm preparing the documentation in RST format to be able to generate documentation on https://readthedocs.org/. This will be the location of our docs. These will be nicely generated and at the same time we will have them properly versioned.

File: docs/descriptor.rst (hunk 4 above)
Comment (dbecvarik) [reply docs/descriptor.rst]: yes but keys are equivalent in  the image descriptor point of view - so its weird to have some in chapters some not - its than hard to say if label is a key from a quick look.

Comment (goldmann) [reply docs/descriptor.rst]: OK, I get the point now, I'll think about it.

**Comment by goldmann** [comment]
All comments should be resolved now.

**Pull request merged**